### PR TITLE
New OD pratical

### PR DESCRIPTION
**Pull request by mem48**

### CODE REVIEW

File: practicals/2-od.md [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+``` r
+nz = spData::nz
+nz_height = spData::nz_height
+```
+
+We can use `tidyverse` functions like `filter` and `select` on `sf`
+objects in the same way you did in Practical 1.
+
+``` r
+canterbury = nz %>% filter(Name == "Canterbury")
```
Comment (Robinlovelace) [inline practicals/2-od.md:111]: Good content from Geocomputation with R.

File: practicals/2-od.Rmd [hunk 2]
```diff
@@ -1,127 +1,178 @@
 ---
 title: "Origin-destination data"
 subtitle: '<br/>Practical'
-author: "Robin Lovelace"
+author: "Malcolm Morgan"
```
Comment (Robinlovelace) [inline practicals/2-od.Rmd:4]:
```suggestion
author: "Malcolm Morgan and Robin Lovelace"
```

File: practicals/2-od.Rmd [hunk 3]
```diff
@@ -1,127 +1,178 @@
...
 ```
 
-# Project set-up and tidyverse testing
+- It relies on **spData**, which loads datasets used in the code examples of this chapter:
+
+```{r 03-attribute-operations-2, echo = T, results = 'hide',warning=FALSE, message=FALSE}
+library(spData)  # spatial data package 
+```
 
-1. Check your packages are up1.to-date with `update.packages()`
```
Comment (Robinlovelace) [inline practicals/2-od.Rmd:45]: :+1: to typo fix

File: practicals/2-od.Rmd [hunk 4]
```diff
@@ -1,127 +1,178 @@
...
+1. Check your packages are up-to-date with `update.packages()`
 1. Create an RStudio project with an appropriate name for this module (e.g. `TDSmodule`)
-1. Create appropriate files for code, data and anything else (e.g. images)
-1. Create a script called `learning-tidyverse.R`, e.g. with the following command:
+1. Create appropriate folders for code, data and anything else (e.g. images)
```
Comment (Robinlovelace) [inline practicals/2-od.Rmd:49]: Do we need this? May be over-complicating this at this stage.

File: practicals/2-od.Rmd [hunk 5]
```diff
@@ -1,127 +1,178 @@
...
-# Project set-up and tidyverse testing
+- It relies on **spData**, which loads datasets used in the code examples of this chapter:
+
+```{r 03-attribute-operations-2, echo = T, results = 'hide',warning=FALSE, message=FALSE}
+library(spData)  # spatial data package 
+```
 
-1. Check your packages are up1.to-date with `update.packages()`
+1. Check your packages are up-to-date with `update.packages()`
 1. Create an RStudio project with an appropriate name for this module (e.g. `TDSmodule`)
```
Comment (Robinlovelace) [inline practicals/2-od.Rmd:48]:
```suggestion
1. Create an RStudio project with an appropriate name for this session (e.g. `practical2`)
```

The old one doesn't really follow from last week when they created a project called `practical`:

https://github.com/ITSLeeds/TDS/blob/28205a0d1d3766a56afae96c876448ef2f5e4601/practicals/1-intro.Rmd#L120

File: practicals/2-od.Rmd [hunk 6]
```diff
@@ -1,127 +1,178 @@
...
 od_data = od_data %>%
   rename(walk = foot)
 ```
 
+Next we will made a new dataset `od_data_walk` by taking `od_data` and piping it (`%>%`) to `filter` the data frame to only include rows where `walk > 0`. Then `select` a few of the columns and calculate two new columns `proportion_walk` and `proportion_drive`.
 
-```{r}
-# transform
+```{r, echo=FALSE}
 od_data_walk = od_data %>%
```
Comment (Robinlovelace) [inline practicals/2-od.Rmd:143]: :+1:

File: practicals/2-od.Rmd [hunk 7]
```diff
@@ -157,117 +208,83 @@ ggplot(od_data_new) +
...
-    alpha = 0.6,
-    col = c("proportion_active", "new_proportion_active"),
-    title = "Active travel (%)"
-  ) +
-  tm_scale_bar()
 
-```
+# Homework
+
+1. Read Chapters 2-5, 7, and 8 of [Geocomputation with R](https://r.geocompx.org/transport.html)
```
Comment (Robinlovelace) [inline practicals/2-od.Rmd:285]: That's quite a lot to get through, I suggest making 7 and 8 bonus questions.

I'd rather everyone did the basics than asking too much and nobody completes the homework.

**Review by Robinlovelace — APPROVED** [review]
Few tweaks and suggestions for you @mem48 otherwise good to go.

File: practicals/2-od.Rmd (hunk 7 above)
Comment (mem48) [reply practicals/2-od.Rmd]: I've made more bonus work

File: practicals/2-od.Rmd (hunk 4 above)
Comment (mem48) [reply practicals/2-od.Rmd:49]: I've left it in, it basic setup that they should be able to do by now

File: practicals/2-od.md (hunk 1 above)
Comment (mem48) [reply practicals/2-od.md:111]: Between R4DS, RRSRR, and Geocomp with R they have a lot of good worked examples to go through. So I don't think we need to create more.

**Pull request merged**